scripts/ubuntu-20-provision.sh

```diff
@@ -110,8 +110,10 @@ rm -rf /tmp/git-lfs*
 apt-get install -y --no-install-recommends fontconfig
 
 ## OpenJDKs: Adoptium - https://adoptium.net/installation.html
+mkdir -p /opt/jdk-8 /opt/jdk-11 /opt/jdk-17
+
 # JDK8
-jdk8_short_version="$(echo "${JDK8_VERSION}" | sed 's/-//g')"
+jdk8_short_version="${JDK8_VERSION//-/}"
```

Comment:
https://stackoverflow.com/a/13210909/12916 TIL, but is there a reason this looks different from the JDK 11 and 17 versions?

Reply:
Outside the non-coherent versioning schemes, I don't know the reason behind their choice.

The code is empirical, based on what we had on the former AdoptOpenJDK download links (and that are the same with Temurin).

You can see the 2 different patterns by looking at https://github.com/jenkins-infra/packer-images/blob/main/.auto.pkrvars.hcl#L3-L5 , and then by checking the download links on https://adoptium.net/ => JDK8 does not have any "separator character" at all, while JDK 11 and 17 have underscores in place of +.